styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -396,7 +397,7 @@ public void create(Environment environment) {
     final DockerRunner dockerRunner = MeteredDockerRunnerProxy.instrument(
         TracingProxy.instrument(DockerRunner.class, routingDockerRunner), stats, time);
 
-    final FlyteRunner flyteRunner = flyteRunnerFactory.create(environment);
+    final FlyteRunner flyteRunner = FlyteRunner.routing(id -> flyteRunnerFactory.create(id, environment), runnerId);
```

Comment:
where is this runnerId generated? I think it's coming from a line 387

```
    final Function<RunState, String> runnerId = new DockerRunnerId(styxConfig);

```

is it what we want here?

Reply:
Ok what we pass to `FlyteRunner.routing` is a lambda that conforms to `FlyteRunner.FlyteRunnerFactory`. This `runnerFactory` get stored in the instant variable and it would be used on 

```java
  protected T runner(RunState runState) {
    var id = runnerId.apply(runState);
    return runners.computeIfAbsent(id, runnerFactory);
  }
```

So the answer is the Id is coming from calling the function `runnerId` over the `RunState`

Reply:
yes, i understand about the runnerFactory.
What about the runnerId lamba. is this the correct one?
Because the one we pass here is this one: https://github.com/spotify/styx/blob/master/styx-scheduler-service/src/main/java/com/spotify/styx/docker/DockerRunnerId.java